app/models/project_association.rb

```diff
@@ -37,7 +37,8 @@ class ProjectAssociation < ActiveRecord::Base
   belongs_to :project_b, class_name:  'Project',
                          foreign_key: 'project_b_id'
 
-  validates_presence_of :project_a, :project_b
+  alias_attribute :dependent_project, :project_b
+  validates_presence_of :project_a, :dependent_project
```

Comment:
Prefer the new style validations `validates :column, presence: value` over `validates_presence_of`.